### PR TITLE
Scheduler and submission config refactoring and cleanup

### PR DESCRIPTION
**Pull request by severinson**
Move config relating to job submission from `SchedulingConfig` into a new struct `SubmissionConfig`. Misc. other config refactoring and cleanup. Improved docstrings/comments.

### CODE REVIEW

File: internal/armada/configuration/types.go [hunk 1]
```diff
@@ -80,12 +85,100 @@ type PulsarConfig struct {
 	ReceiverQueueSize int
 }
 
+// SubmissionConfig contains config relating to job submission.
+type SubmissionConfig struct {
```
Comment (severinson) [inline internal/armada/configuration/types.go:89]: New struct for config relating to job submission.

File: internal/armada/configuration/types.go [hunk 2]
```diff
@@ -80,12 +85,100 @@ type PulsarConfig struct {
...
+// 3. Gang scheduling, optional across clusters, and with lower and upper bounds on the number of jobs scheduled.
+//
+// Note that Armada still relies on kube-scheduler for binding of pods to nodes.
+// This is achieved by adding to each pod created by Armada a node selector that matches only the intended node.
 type SchedulingConfig struct {
 	// Set to true to disable scheduling
 	DisableScheduling bool
 	// Set to true to enable scheduler assertions. This results in some performance loss.
 	EnableAssertions bool
-	Preemption       PreemptionConfig
```
Comment (severinson) [inline internal/armada/configuration/types.go:88]: Preemption is so integrated it doesn't make sense to keep the config in a separate struct anymore.

File: internal/armada/configuration/types.go [hunk 3]
```diff
@@ -80,12 +85,100 @@ type PulsarConfig struct {
...
+// The Armada scheduler is part of the Armada control plane.
+//
+// Features:
+// 1. Queuing and fairly dividing resources between users.
+// 2. Fair preemption, including between jobs of equal priority to balance resource allocation.
+// 3. Gang scheduling, optional across clusters, and with lower and upper bounds on the number of jobs scheduled.
+//
+// Note that Armada still relies on kube-scheduler for binding of pods to nodes.
+// This is achieved by adding to each pod created by Armada a node selector that matches only the intended node.
 type SchedulingConfig struct {
```
Comment (severinson) [inline internal/armada/configuration/types.go:150]: I've moved config relating to submission from here to the new struct.

File: config/executor/config.yaml [hunk 4]
```diff
@@ -31,6 +31,8 @@ kubernetes:
   impersonateUsers: false
   trackedNodeLabels:
     - kubernetes.io/hostname
+  toleratedTaints:
```
Comment (d80tb7) [inline config/executor/config.yaml:34]: do we want this?

Comment (severinson) [reply config/executor/config.yaml]: I've removed it.

**Review by d80tb7 — APPROVED** [review]

**Pull request merged**